jaxopt/_src/lbfgs.py

```diff
@@ -255,7 +264,9 @@ def init_state(self,
                       s_history=init_history(init_params, self.history_size),
                       y_history=init_history(init_params, self.history_size),
                       rho_history=jnp.zeros(self.history_size, dtype=dtype),
-                      aux=aux)
+                      gamma=jnp.asarray(1.0, dtype=dtype),
+                      aux=aux,
+                      failed=jnp.asarray(False))
```

Comment:
I'm not sure we need `asarray` for booleans. CC @froystig

Reply:
I can try without and tell you

Reply:
it works without the `asarray`, so removed it

Reply:
If we want them as explicit scalar arrays, rather than Python scalars, then `np.array(False)` where `np` is original numy. If we moreover want them on device for some reason, then `jnp.asarray(False)`.

Reply:
Here the main thing we want to be careful of is that `init_state` and `update` output states with the same types. Otherwise, it triggers recompilations.

Reply:
I believe we're fine with just a plain literal `False`:
```python
>>> import jax
>>> import jax.numpy as jnp
>>> @jax.jit
... def f(x):
...   print('compiling')
...   return x
... 
>>> f(False)
compiling
DeviceArray(False, dtype=bool)
>>> f(jnp.asarray(False))
DeviceArray(False, dtype=bool)
```

Reply:
this would mean rewriting the unit tests that check the return types, so maybe it's for a different PR?

Reply:
Yes, I agree. Let's squash the commits and merge this PR.